python/whylogs/api/store/local.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import os
+from datetime import datetime
+from functools import reduce
+from glob import glob
+from typing import Optional
+
+import whylogs as why
+from whylogs.api.store.date_config import DateConfig
+from whylogs.api.store.profile_store import ProfileStore
+from whylogs.api.store.utils.random_strings import random_string
+from whylogs.api.writer.local import LocalWriter
+from whylogs.core import DatasetProfileView
+
+DEFAULT_DIR = "profile_store/"
+logger = logging.getLogger(__name__)
+
+
+class LocalStore(ProfileStore):
+    """
+    The LocalStore is the implementation of the base :class:ProfileStore
+    that will manage reading and writing profiles on the local file system.
+
+    In order to instantiate the object, you will need to define a base_name,
+    which is related to the name of the model or dataset you're profiling.
+    To properly use the LocalStore to generate files, you should append it
+    to your existing Rolling Logger, as the below example demonstrates.
+
+    ```python
+    import whylogs as why
+    from whylogs.api.store.local import LocalStore
+
+    logger = why.logger(mode="rolling", interval=10, when="S")
+    logger.append_store(store=LocalStore(base_name="my_model"))
+
+    new_df = model.predict(input_data)
+    logger.log(new_df)
+    ```
+
+    The above piece of code will make sure to write the logged profiles
+    to the correct location that can be further fetched using the same
+    LocalStore, like so:
+
+    ```python
+    from datetime import datetime, timedelta
+
+    from whylogs.api.store.date_config import DateConfig
+
+    store = LocalStore(base_name="my_model")
+    date_config = DateConfig(
+        start_date = datetime.utcnow() - timedelta(days=7),
+        end_date = datetime.utcnow()
+    )
+
+    profile_view = store.get(date_config=date_config)
+    ```
+
+    This will fetch all existing profiles from `my_model` from the past
+    7 days in a single merged DatasetProfileView.
+
+    :param base_name: the name of the dataset or model that you're logging
+    :type base_name: str
+
+    """
+
+    def __init__(self, base_name: str):
+        self.base_name = base_name
+        self._default_path = os.path.join(os.getcwd(), DEFAULT_DIR)
+        logger.debug(f"Default ProfileStore path set to {self._default_path}")
+        self._writer = LocalWriter(base_dir=self._default_path)
+
+        if not os.path.isdir(os.path.join(self._default_path, self.base_name)):
+            os.makedirs(os.path.join(self._default_path, self.base_name))
+
+    @staticmethod
+    def _get_profile_timestamp() -> str:
+        now = datetime.utcnow()
+        return f"profile_{now.date()}_{now.hour}:{now.minute}:{now.second}_{random_string()}.bin"
+
```

Comment:
More like `_get_profile_filename` right? Might be a python question here, why bother making private methods static rather than just making this a function in the module above?

Reply:
renamed it! 
about the private vs. function, i rather keep this method inside the class because I only want local store to use this, not other stuff. and since it doesn't rely on any of the object's attrs, i made it static.